plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp

```diff
@@ -973,25 +972,31 @@ class read_write {
    read_write(controller& db, const fc::microseconds& abi_serializer_max_time, bool api_accept_transactions);
    void validate() const;
 
-   using push_block_params = chain::signed_block_v0;
+   using push_block_params_v1 = chain::signed_block_v0;
    using push_block_results = empty;
-   void push_block(push_block_params&& params, chain::plugin_interface::next_function<push_block_results> next);
+   void push_block(push_block_params_v1&& params, chain::plugin_interface::next_function<push_block_results> next);
 
-   using push_transaction_params = fc::variant_object;
+   using push_transaction_params_v1 = fc::variant_object;
    struct push_transaction_results {
       chain::transaction_id_type  transaction_id;
       fc::variant                 processed;
    };
-   void push_transaction(const push_transaction_params& params, chain::plugin_interface::next_function<push_transaction_results> next);
-
+   void push_transaction(const push_transaction_params_v1& params, chain::plugin_interface::next_function<push_transaction_results> next);
 
-   using push_transactions_params  = vector<push_transaction_params>;
+   using push_transactions_params_v1  = vector<push_transaction_params_v1>;
    using push_transactions_results = vector<push_transaction_results>;
-   void push_transactions(const push_transactions_params& params, chain::plugin_interface::next_function<push_transactions_results> next);
+   void push_transactions(const push_transactions_params_v1& params, chain::plugin_interface::next_function<push_transactions_results> next);
 
-   using send_transaction_params = push_transaction_params;
+   using send_transaction_params_v1 = push_transaction_params_v1;
    using send_transaction_results = push_transaction_results;
-   void send_transaction(const send_transaction_params& params, chain::plugin_interface::next_function<send_transaction_results> next);
+   void send_transaction(const send_transaction_params_v1& params, chain::plugin_interface::next_function<send_transaction_results> next);
+
+   struct send_transaction_params_v2 {
+      bool return_failure_traces = false;
```

Comment:
We want `return_failure_traces = true` default for develop-boxed.

Reply:
for both v1 and v2?

Reply:
yes

Reply:
done